src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts

```diff
@@ -63,6 +68,42 @@ export const selectStakingAccountIds = createDeepEqualOutputSelector(
   (byAccountId): AccountId[] => Object.keys(byAccountId),
 )
 
+export const selectUserStakingOpportunitiesWithMetadataByFilter = createSelector(
+  selectUserStakingOpportunitiesById,
+  selectStakingOpportunitiesById,
+  selectAccountIdParamFromFilter,
+  selectAssetIdParamFromFilter,
+  (
+    userStakingOpportunitiesById,
+    stakingOpportunitiesById,
+    accountId,
+    assetId,
+  ): UserStakingOpportunityWithMetadata[] =>
+    Object.entries(userStakingOpportunitiesById)
+      .filter(([userStakingId]) => {
+        const [userStakingAccountId, stakingId] = deserializeUserStakingId(
+          userStakingId as UserStakingId,
+        )
+
+        return (
+          (!accountId || accountId === userStakingAccountId) &&
+          (!assetId || assetId === (stakingOpportunitiesById[stakingId]?.assetId ?? ''))
+        )
+      })
+      .map(([userStakingId, userStakingOpportunity]) => {
+        const [, stakingId] = deserializeUserStakingId(userStakingId as UserStakingId)
+        if (!stakingOpportunitiesById[stakingId] || !userStakingOpportunity) return undefined
+
+        const userStakingOpportunityWithMetadata = {
+          ...stakingOpportunitiesById[stakingId],
+          ...userStakingOpportunity,
+        } as UserStakingOpportunityWithMetadata
```

Comment:
asssiiiiignment over cast?

Reply:
discrimination is sad 😢 
<img width="1638" alt="image" src="https://user-images.githubusercontent.com/17035424/217467743-8bd967cd-4d9e-4dc5-8b58-387914663a23.png">

Reply:
it feels like this is the root cause

https://github.com/shapeshift/web/blob/feat_wire_up_cosmos_opportunities_things/src/state/slices/opportunitiesSlice/types.ts#L60

should be intersection (`&`) rather than union (`|`) yeah?